server/app/services/applications/ProgramAdminApplicationService.java

```diff
@@ -238,6 +241,53 @@ public Optional<ApplicationModel> getApplication(long applicationId, ProgramDefi
     }
   }
 
+  /*
+   * Retrieves the applications for the give ApplicationIds and validates that it is associated with the given
+   * program.
+   */
+  public ImmutableList<ApplicationModel> getApplications(
```

Comment:
Blocking: Can we add unit tests for this method?

Reply:
Sure, I have created #9015 to track it. I will be addressing it in the next PR.